drivers/cifutils.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/python
+#
+# Copyright (C) Citrix Systems Inc.
+#
+# This program is free software; you can redistribute it and/or modify 
+# it under the terms of the GNU Lesser General Public License as published 
+# by the Free Software Foundation; version 2.1 only.
+#
+# This program is distributed in the hope that it will be useful, 
+# but WITHOUT ANY WARRANTY; without even the implied warranty of 
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# ISOSR: remote iso storage repository
+
+import SR, VDI, SRCommand, util
+import nfs
+import os, re
+import xs_errors
+
+
+def getCIFCredentials(dconf, session):
+    credentials={}
+    if dconf.has_key('username') \
+            and (dconf.has_key('cifspassword') or dconf.has_key('cifspassword_secret')):
+        dom_username = dconf['username'].split('\\')
+        if len(dom_username) == 1:
+            domain = None
+            username = dom_username[0]
+        elif len(dom_username) == 2:
+            domain = dom_username[0]
+            username = dom_username[1]
+        else:
+            err_str = ("A maximum of 2 tokens are expected "
+                       "(<domain>\<username>). {} were given."
+                       .format(len(dom_username)))
+            util.SMlog('CIFS credentials error: ' + err_str)
+            raise xs_errors.XenError('ISOMountFailure', opterr=err_str)
+
+        if dconf.has_key('cifspassword_secret'):
+            password = util.get_secret(session,dconf['cifspassword_secret'])
+        else:
+            password = dconf['cifspassword']
+
+        domain = util.to_plain_string(domain)
+
+        credentials["username"] = util.to_plain_string(username)
```

Comment:
as the keys of this will map directly to the environment variables they need to be USER and PASSWD. Not sure how the domain gets handled.

Reply:
Looks like domain cannot be passed by environment, which is really irritating. However it's less of a problem to pass domain via the command line option to `mount`.

Reply:
doh! Fixed